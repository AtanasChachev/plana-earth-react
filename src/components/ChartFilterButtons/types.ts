export interface ChartFilterButtonsProps {
  onClick(interval: string): void;
}

export interface Return {
  activeIndex: number;
  handleFilterButton: (interval: string, index: number) => void;
}