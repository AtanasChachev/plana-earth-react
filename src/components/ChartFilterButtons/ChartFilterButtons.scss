@import 'styles/utils/mixins';

.chart-buttons {
  display: inline-block;
  border-width: 0 0 1px;
  border-style: solid;
  border-color: var(--clr-caribbean-green);
  margin-bottom: 20px;

  @include breakpoint(max, width, 767px) {
    display: flex;
  }

  &__button {
    &.MuiButton-root.MuiButton-root {
      padding: 6px 20px;
      border-width: 0 1px 0 0;
      border-radius: 0;
      font-weight: 400;

      @include breakpoint(max, width, 767px) {
        font-size: 12px;
        min-width: 60px;

        &:last-of-type {
          border: 0;
        }
      }
    }

    &--active {
      background-color: var(--clr-caribbean-green) !important;
      color: var(--clr-white) !important;
    }
  }
}