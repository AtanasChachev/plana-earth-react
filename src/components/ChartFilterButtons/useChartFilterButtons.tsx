import { useState } from 'react';
import { ChartFilterButtonsProps } from './types';
import { Return } from './types';

export const useChartFilterButtons = ({ onClick }: ChartFilterButtonsProps): Return => {
  const [activeIndex, updateActiveIndex] = useState<number>(0);

  const handleFilterButton = (interval: string, index: number) => {
    updateActiveIndex(index);
    onClick(interval);
  };

  return {
    activeIndex,
    handleFilterButton,
  };
};