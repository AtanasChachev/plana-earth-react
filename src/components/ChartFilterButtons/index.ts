import ChartFilterButtons from './ChartFilterButtons';

export default ChartFilterButtons;