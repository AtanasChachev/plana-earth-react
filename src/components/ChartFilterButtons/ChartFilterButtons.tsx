import { Button } from '@material-ui/core';
import { capitalizeWords } from 'utils/helpers';
import { CHART_SETTINGS  } from 'constants/index';
import { useChartFilterButtons } from './useChartFilterButtons';

import './ChartFilterButtons.scss';

export interface ChartFilterButtonsProps {
  onClick(interval: string): void;
}

const ChartFilterButtons = ({ onClick }: ChartFilterButtonsProps): JSX.Element => {
  const { activeIndex, handleFilterButton } = useChartFilterButtons({ onClick });

  return (
    <div className="chart-buttons">
      {
        CHART_SETTINGS.chartFilterButtons.map((interval: string, index: number) => {
          const handleClick = () => handleFilterButton(interval, index);

          return (
            <Button
              data-testid="chart-filter-button"
              aria-label={`Chart filter button - click to filter the chart by ${interval}`}
              key={index} 
              className={`chart-buttons__button ${index === activeIndex ? 'chart-buttons__button--active' : ''}`}
              color="primary"
              onClick={handleClick}>{capitalizeWords(interval)}
            </Button>);
        },
        ) 
      }
    </div>
  );
};

export { ChartFilterButtons };