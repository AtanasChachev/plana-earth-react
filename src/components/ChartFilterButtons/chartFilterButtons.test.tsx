import ChartFilterButtons from './ChartFilterButtons';
import { render, screen, act, fireEvent } from '@testing-library/react';

const onClick = jest.fn();

const setup = () => {
  const { asFragment } = render(<ChartFilterButtons onClick={onClick} />);
  const filterButton = screen.getAllByTestId('chart-filter-button')[0];

  return { 
    filterButton,
    asFragment,
  };
};

describe('<ChartFilterButtons />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('<ChartFilterButtons /> should render correctly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });

  test('ChartFilterButtons button should emit click', () => {
    const { filterButton } = setup();

    act(() => {
      fireEvent.click(filterButton);
    });

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});