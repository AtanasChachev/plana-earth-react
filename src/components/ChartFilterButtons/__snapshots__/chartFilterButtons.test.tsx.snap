// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartFilterButtons /> <ChartFilterButtons /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chart-buttons"
  >
    <button
      aria-label="Chart filter button - click to filter the chart by day"
      class="MuiButtonBase-root MuiButton-root MuiButton-text chart-buttons__button  MuiButton-textPrimary"
      data-testid="chart-filter-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Day
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Chart filter button - click to filter the chart by week"
      class="MuiButtonBase-root MuiButton-root MuiButton-text chart-buttons__button  MuiButton-textPrimary"
      data-testid="chart-filter-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Week
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Chart filter button - click to filter the chart by month"
      class="MuiButtonBase-root MuiButton-root MuiButton-text chart-buttons__button  MuiButton-textPrimary"
      data-testid="chart-filter-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Month
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Chart filter button - click to filter the chart by quarter"
      class="MuiButtonBase-root MuiButton-root MuiButton-text chart-buttons__button  MuiButton-textPrimary"
      data-testid="chart-filter-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Quarter
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="Chart filter button - click to filter the chart by year"
      class="MuiButtonBase-root MuiButton-root MuiButton-text chart-buttons__button  MuiButton-textPrimary"
      data-testid="chart-filter-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Year
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
